clang15 prevents automatic conversion of `void *` to `u64`

When [`map.call()`](https://github.com/iovisor/bcc/blob/master/docs/reference_guide.md#27-mapcall) appears in the code, an implicit type conversion from `void*` to `u64` is required, but this is not allowed in clang15.
```
/virtual/main.c:24:20: error: incompatible pointer to integer conversion passing 'void *' to parameter of type 'u64' (aka 'unsigned long long') [-Wint-conversion]
    bpf_tail_call_((void *)bpf_pseudo_fd(1, -1), ctx, 2);
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/virtual/include/bcc/helpers.h:552:25: note: passing argument to parameter 'map_fd' here
void bpf_tail_call_(u64 map_fd, void *ctx, int index) {
                        ^
1 error generated.
Traceback (most recent call last):
  File "/home/admin/Templates/tmp.rhOt4STjdi/1.py", line 2, in <module>
    b = BPF(src_file="example.c")
  File "/usr/lib/python3.10/site-packages/bcc/__init__.py", line 479, in __init__
    raise Exception("Failed to compile BPF module %s" % (src_file or "<text>"))
Exception: Failed to compile BPF module b'example.c'
```


**To reproduce the problem**:
You need to use clang15. Then run the `map.call()` example in the [documentation](https://github.com/iovisor/bcc/blob/master/docs/reference_guide.md#27-mapcall).


Although I couldn't find this change in the clang15 changelog (maybe I didn't look carefully), it is indeed not allowed.
[Here](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1AB9U8lJL6yAngGVG6AMKpaAVxYMJATlIOAMngMmABy7gBGmMR6AA6oCoS2DM5uHjFxCQIBQaEsEVFcFphWNgJCBEzEBMnunlw%2BlpjWiWUVBFkh4ZF6CuWV1akFPa3tOXkSAJQWqK7EyOwcAKQATADMgchuWAsrjj34ggB0CNvYCxoAggBuqHjoANRUrgzIEK6BBABsksYEd3dM4wWAHYAEJnc5/YiYAgzBjbMEXYEAEXB7zuLCYgQggNB4L%2Bb0EXx%2Bd1U2yRXHheIeTxeEGutwAVONVICVgjzsjwRxJrROABWXieDhaUioTiOO4KaazTB3ZYrHikAiabmTADWICWGn0nEkgpVos4vAUIG1yuF3NIcFgSDQLGidEi5Eodod9CiyDC0SoG0MwC4fI02qwlzwcwAanhMAB3ADy0UYnEVNFoBEiJogYQNYUCFQAnkneDnmMQ87GwtpGubFXa2IJYwxaAWLaQsBijOIW/goU1LpgTS3MKpGq404XyIIigbaHgwsR884sAaCMQ8CxC5MqAZgApIzH44nuLx%2BIIRGJ2FIZIJFCp1C3dAUDEYQKZjOYZ2ETZBJqhoiUGAOxpFFWiT2AwTguDUej%2BIEHS5F0BSxPE/79LUpBIRkDAjJ0%2BSFMUzS9FUkEDHhIGlIR2HwbhQx9MRaE0W0sGjAhkxSjMcwTDqHACqQQoimKHB3CCygAGJ3L6Rh3AGBwaDJdwQLghAkHKqxcOMvDmlo4zqpq2q8hweq8QaAnGqaSoqpM1qICA0wENEo7OhArqOsQwSsPMwliRJwBSXyMkybwmD4EQq7oHoJ7CKI4iXhFN5qAaD6kNG87RBuXE8XxvACbGo72b8qBUEJoniU%2BPnSbJGjyc49ouSpCrqeZFraaQGpalxhmZYaHCmWaFlWjA1nOe6jlDV0nret5UhBq2mChhGUZxgmQrJnQabEBmWYtsW%2BbjttpblpW1jjrWjAEA2TYGm2fqdiK3YgX2A4ikOI5jkeE5pvpIofvOpaLvMIormuaVbkwO57oth7JrIZ7RdIsVKPF96avofovmY%2Bizl%2B2Kin%2BiSAaRTR2BADioRIKy%2BOBlFjEC6HpChdFk7TyGJFTXQ0w0hMMC0tEpLU5Mc/%2B3OMdkOEgOzhGk1w/MUUxotAqx0ocWp6X6i2AlFV5pVSZI/mVQpwXKfKakaRZOltfpHXGUaFhmZplpWRASC2XlI2oDV7puWwnCeSVfra7rgUG6F4XQ1FF5w7IcV3iKuhLElKVpfpGVWxwOV2aOdwFRrvuSVIutVW7bqRHVSwNXbzWtXpuqq/x1smr1TU8pwSw11l1vl5MfbraBkhAA%3D%3D%3D) is a comparison of the output of clang14 and clang15 for a similar problem. You can see that in clang14, that's just a warning. But in the output of clang15, we get an error.

--- a/src/cc/export/helpers.h
+++ b/src/cc/export/helpers.h
@@ -549,8 +549,8 @@ static int (*bpf_probe_read_str)(void *dst, u64 size, const void *unsafe_ptr) =
   (void *) BPF_FUNC_probe_read_str;
 int bpf_trace_printk(const char *fmt, ...) asm("llvm.bpf.extra");
 static inline __attribute__((always_inline))
-void bpf_tail_call_(u64 map_fd, void *ctx, int index) {
-  ((void (*)(void *, u64, int))BPF_FUNC_tail_call)(ctx, map_fd, index);
+void bpf_tail_call_(void *map_fd, void *ctx, int index) {
+  ((void (*)(void *, u64, int))BPF_FUNC_tail_call)(ctx, (u64)map_fd, index);
 }
 static int (*bpf_clone_redirect)(void *ctx, int ifindex, u32 flags) =
   (void *) BPF_FUNC_clone_redirect;

